core/iwasm/libraries/lib-socket/inc/wasi_socket_ext.h

```diff
@@ -140,6 +140,9 @@ struct addrinfo {
     struct addrinfo *ai_next; /* Pointer to next in list.  */
 };
 
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
A little strange that the macro only covers part of the code, prefer to wrap the whole file:
```C
#ifndef _WASI_SOCKET_EXT_H_
#define _WASI_SOCKET_EXT_H_

#include <stdbool.h>
#include <stddef.h>
#include <stdint.h>

#ifdef __cplusplus
extern "C" {
#endif
   => leave an empty line
...
   => leave an empty line
#ifdef __cplusplus
}
#endif
   => leave an empty line
#endif /* end of _WASI_SOCKET_EXT_H_ */
```